redwood/src/decryption.rs

```diff
@@ -58,20 +60,25 @@ impl<'a> DecryptionHelper for Helper<'a> {
             .key()
             .clone();
 
-        // Decrypt the secret key with the specified passphrase.
-        let mut pair = key.decrypt_secret(self.passphrase)?.into_keypair()?;
+        for pkesk in pkesks {
+            if pkesk.recipient() == &KeyID::from(key.fingerprint()) {
+                // Decrypt the secret key with the specified passphrase.
+                let mut pair =
+                    key.decrypt_secret(self.passphrase)?.into_keypair()?;
+                pkesk
+                    .decrypt(&mut pair, sym_algo)
+                    .map(|(algo, session_key)| decrypt(algo, &session_key));
+                // XXX: The documentation says:
+                // > If the message is decrypted using a PKESK packet, then the
+                // > fingerprint of the certificate containing the encryption subkey
+                // > should be returned. This is used in conjunction with the intended
+                // > recipient subpacket (see Section 5.2.3.29 of RFC 4880bis) to
+                // > prevent Surreptitious Forwarding.
+                // Unclear if that's something we need to do.
+                return Ok(None);
```

Comment:
:thinking: It should be as easy as `return Ok(key.fingerprint());` ...

Reply:
Yeah, it only clicked sometime last night that this was trivial to do, added to the PR.